airflow/migrations/versions/0032_3_0_0_drop_execution_date_unique.py

```diff
@@ -44,14 +45,24 @@
 
 def upgrade():
     with op.batch_alter_table("dag_run", schema=None) as batch_op:
-        batch_op.alter_column("execution_date", new_column_name="logical_date", existing_type=sa.TIMESTAMP)
+        batch_op.alter_column(
+            "execution_date",
+            new_column_name="logical_date",
+            existing_type=UtcDateTime,
```

Comment:
What we have in execution_date is UtcDatetime before AF 3. This was changed when execution_date was changed to logical_date. I don't think we should change to Timestamp. Also, all of our date columns currently use UtcDatetime

Reply:
Those are the same. Both use `TIMESTAMP`

https://github.com/apache/airflow/blob/5b2a96ee72a4bd7b7fbd4925c2f8c8468f047a64/airflow/utils/sqlalchemy.py#L48-L62

https://github.com/apache/airflow/blob/5b2a96ee72a4bd7b7fbd4925c2f8c8468f047a64/airflow/migrations/db_types.py#L43-L68

The difference IMO lies in `UtcDateTime` is mainly used in the Application/ORM layer while `db_types.TIMESTAMP` should be used for alembic migration.

Reply:
I had some tests failing when the migration use TIMESTAMP instead of UtcDatetime. The tests under trigger_dagrun operator. I'll change it to TIMESTAMP to reveal those tests. 
I believe the reason those tests fail is that in this PR, the test database is initialized with the migration files which picks up the TIMESTAMP

Reply:
Maybe it is worth consolidating them if it is possible

Reply:
I imported the Timestamp from db_types instead of sa.TIMESTAMP. Let's see how it goes

Reply:
Went well!